### PR TITLE
SP-GiST index on geometry

### PR DESCRIPTION
**Pull request by Komzpa**
Follow-up PR from https://github.com/postgis/postgis/pull/251

@alesuiss @estebanzimanyi

### CODE REVIEW

File: postgis/postgis.sql.in [hunk 1]
```diff
@@ -690,7 +690,7 @@ CREATE OR REPLACE FUNCTION geometry_overright(geom1 geometry, geom2 geometry)
 -- Availability: 0.1.0
 CREATE OPERATOR &> (
 	LEFTARG = geometry, RIGHTARG = geometry, PROCEDURE = geometry_overright,
-	COMMUTATOR = '&<',
+--	COMMUTATOR = '&<',
```
Comment (Komzpa) [inline postgis/postgis.sql.in]: @alesuiss this was fixed in trunk, use `git merge upstream/svn-trunk`

Comment (alesuiss) [reply postgis/postgis.sql.in]: Thanks :-)

We did install travis on our fork to test it there, but used PG 9.6 so didn't compile.

**Comment by strk** [comment]
Please remember about upgrades when fixing these things

**Comment by alesuiss** [comment]
@strk can you elaborate? What should we do?

**Comment by strk** [comment]
Enable an old version of PostGIS in a database,
the upgrade it to the new version, then check
if the database objects of PostGIS (like operators,
with their commutators) are the same in the so-upgraded
database and in a database where the NEW PostGIS was
installed right away.

Adding an automated testcase for the presence of such
commutators should be enough because the testing framework
does run some upgrade tests (at least the Drone one, which
is only run when the code is merged OR with some more work
when code is pushed to the OSGeo Git hosting facility)

**Comment by alesuiss** [comment]
@strk actually those commutators were fixed upstream by Paul Ramsey, but we also implemented the fix on our side as it was needed for correct results. This lead to tree conflict, which I resolved manually before @Komzpa suggested to simply merge svn-trunk.

**Comment by alesuiss** [comment]
Hi,

FYI, I've been fiddling with Travis to get PG11 working. You can see the branch [here](https://github.com/ULB-CoDE-WIT/postgis/blob/travis-pg11/.travis.yml) and the latest results [here](https://travis-ci.com/ULB-CoDE-WIT/postgis/builds/75740247).

There are a few outstanding issues:
- For some reason, it insists on building with clang (3.9), which doesn't recognize -Og 
- One of the tests, "tickets", fails by apparently crashing pg. Curiously, that test passes fine on my workstation. Then, subsequent tests fail because pg is still in recovery mode.

**Comment by robe2** [comment]
Is the associated ticket for this - https://trac.osgeo.org/postgis/ticket/1847

**Comment by robe2** [comment]
Is https://github.com/postgis/postgis/pull/251 already incorporated into this pull request?

**Comment by alesuiss** [comment]
@robe2 This PR is a generalization of #251 to work on the geometry type using the new compress function in PG11.

**Comment by robe2** [comment]
@alesuiss @estebanzimanyi  need your names and company if any for Credits.

I'm still working on ifdef logic for this .  I tested against 11beta2 and all passed.

Some things look like they could go into lower versions, but I'm just going to ifdef all to require PG 11.

**Comment by robe2** [comment]
I noticed in postgis/gserialized_gist_2d.c only change was removal of static on a bunch of functions.  Is this change related to the purpose of this commit or is it unintentional.

**Comment by alesuiss** [comment]
> I noticed in postgis/gserialized_gist_2d.c only change was removal of static on a bunch of functions. Is this change related to the purpose of this commit or is it unintentional.

@robe2 we removed the statics because we need to call these functions for indexes that we are implementing in a separate project. This PR is actually spun off that project.

**Comment by robe2** [comment]
@alesuiss  okay so I'll leave that out then.  If we change to that it would need to come in as a separate pull request and ticket.

What about my other question name for credits?

**Comment by alesuiss** [comment]
> What about my other question name for credits?

We are Esteban Zimányi and Arthur Lesuisse from Université Libre de Bruxelles (ULB). Thanks :-)

**Comment by robe2** [comment]
@Komzpa   did you say you were going to do the documentation for this?

Normally I don't accept pull requests for new features unless they include user documentation of the feature. In this case since  I'm in a rush to release 2.5.0beta1 I did.  No super rush on doc.  End of July is fine.  @alesuiss , @estebanzimanyi   feel free to help @Komzpa  with the documentation work.

You can pattern it after the GIST and BRIN ones  in this section:

https://postgis.net/docs/manual-dev/using_postgis_dbmanagement.html#gist_indexes

**Pull request closed without merging**